lib/kubeclient.rb

```diff
@@ -202,5 +202,11 @@ def all_entities
         result_hash[key_name] = send(method_name)
       end
     end
+
+    def server_api_versions
+      versions_endpoint = URI.parse(@api_endpoint.to_s)
+      versions_endpoint.path.slice! @api_version
+      RestClient::Resource.new(versions_endpoint.to_s).get
```

Comment:
2 things

- I don't think using .slice! on an method on URI#path is a good idea.  I recommend using `versions_endpoint.path = versions_endpoint.path.slice(@api_version)`.
- Should this be wrapped in `handling_kube_exception`
- I'm not sure on the name of the method.  I was thinking just `.api`.  I'm concerned that if in the future they add other stuff besides "versions" to the api endpoint, then this method name has to change, or will be inaccurate.  ".api" implies that it's the data from the api endpoint.